http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -807,6 +815,103 @@ public void testHttpStatusListenerException()
         executeRequest(request, createStatusResponseHandler());
     }
 
+    public record Thing(String s, int i) {}
+
+    @Test
+    public void testStreamingJson()
+            throws Exception
+    {
+        int qty = 25;
+
+        List<Thing> things = IntStream.range(0, qty).mapToObj(i -> new Thing(Integer.toString(i), i)).collect(toImmutableList());
+        String json = new ObjectMapperProvider().get().writeValueAsString(things);
+        JsonCodec<Thing> codec = jsonCodec(Thing.class);
+
+        servlet.setResponseBody(json);
+        servlet.addResponseHeader("Content-Type", "application/json");
```

Comment:
I wonder if support for `application/x-ndjson` is something to consider

Reply:
We can have a new `StreamingResponseHandler` for that, adding it when Airlift's other JSON hander gets support.